### ISSUE TITLE
How Can I go back to Bolt.new once I connect my project with github after that Open in. bolt.new AI option hides.

### ISSUE THREAD

**Issue by SheikhSaad** [stackblitz/bolt.new#688]
**Is your feature request related to a problem? Please describe:**
Going back to **Open in. bolt.new AI**  option hides from the project once I connect my app with github.

<img width="588" alt="image" src="https://github.com/user-attachments/assets/442b0036-56a9-4d2f-af97-ca635117b06b">

**Describe the solution you'd like:**

There should be option to open project in bolt.new again.

<img width="282" alt="image" src="https://github.com/user-attachments/assets/c50f50a4-d3a9-4fc5-8917-2a9c832628aa">

**Describe alternatives you've considered:**

I use my old project from bolt.new and open that in stackblitz and after that save in a create new repository again, so each time changes in bolt.new  I need to create new repository.

**Comment by neg-0** [comment]
I'm experiencing the same issue. Creating a GitHub repository from the StackBlitz editor seems to lose the connection between that repo and the bolt project.

**Comment by jim80** [comment]
This seems to be the way it works right now

**Comment by profplum700** [comment]
What I did was, from my bolt.new chat, clicked the _Open in StackBlitz_ button, then inside StackBlitz clicked the _Create a repository_ button.

<img width="959" alt="image" src="https://github.com/user-attachments/assets/d34f101c-292c-4236-bcb7-cf2ce8b3a050">

It creates the repository but that never changes to something like "View repository".

It's frustrating because I'd like to edit in another code editor too (Cursor AI) from time to time. Be nice to have everything synced between them.

**Comment by jhosefmax** [comment]
I also would like to open an existent project in bolt.new and it seems to not be possible. My github account is already connected on Stackblitz. I can open my projects in Stackblitz but not in bolt.new. Am I missing something?

**Comment by peppix2010** [comment]
This is a really important issue considering in any SDLC you should be able to push your changes, test them and then go back to do some improvements or new features

**Comment by jim80** [comment]
adding git to the virtual machine used would be good, even without any gui in the editor. How easy would be be to *just* do that? :)

**Comment by cmacpro** [comment]
Same Question here.

**Comment by itsmikeferrari** [comment]
I can import my project if it's public, but not a private repo.

The URL is https://bolt.new/~/github.com/yourGithubHandle/yourRepoName

**Comment by jim80** [comment]
@itsmikeferrari - that works, also for branches. I had no idea! Thank you very much!
WHO KNEW!!!!!! :)

**Comment by divparty** [comment]
Just thought I would share that I was able to continue editing a **private** repo in Bolt by going to the repo ON GitHub then you will see this link in the top right which will allow you to edit it again on Bolt, hope this helps someone until they integrate it more seamlessly into their interface.
![image](https://github.com/user-attachments/assets/b95af18c-6df2-4ee0-a768-04122b88128e)

**Comment by tomwojcik** [comment]
@divparty but it doesn't fetch the latest changes that were made outside of the bolt project. It literally loads the existing one, without making it possible to fetch changes. It's the equivalent of opening bolt.new and loading a project from the left sidebar.

Disappointing.

**Comment by johhansantana** [comment]
would love to get this feature. I tried going into the bolt new url by passing the github repository in the url. It tries to import it but fails even though I have stackblitz connected to my repository.

**Comment by benjamincano** [comment]
I can open a new project if its public not private.

**Comment by benjamincano** [comment]
Go to https://stackblitz.com/ and in Recent projects click on the 3 dots at the right and select open in Bolt.new, this worked for me as the other solutions mentioned here didn't.

**Comment by uCRFT** [comment]
> What I did was, from my bolt.new chat, clicked the _Open in StackBlitz_ button, then inside StackBlitz clicked the _Create a repository_ button.
> 
> <img alt="image" width="959" src="https://private-user-images.githubusercontent.com/13546520/377766085-d34f101c-292c-4236-bcb7-cf2ce8b3a050.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyMDg2ODUsIm5iZiI6MTczMDIwODM4NSwicGF0aCI6Ii8xMzU0NjUyMC8zNzc3NjYwODUtZDM0ZjEwMWMtMjkyYy00MjM2LWJjYjctY2YyY2U4YjNhMDUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI5VDEzMjYyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczZmVhZmRkMzVjMmZmMTk3OTQzZjVlMDM1MjljNDg1MDAwMDc4MDI4MmViZmM4ODA0MzE0ZjRkM2JkNzYyZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7f44cL6l0Y_JvK2rTUiZftBtiAyrL3SQtFE48gN6k_0">
> It creates an empty repository and the button never changes to something like "View repository".
> 
> It's frustrating 

same issue, whatever the browser.
And when you change browser, the boltnew chat history is not shared either, it creates a complete different chat history.
Interesting but why ?

**Comment by johhansantana** [comment]
there is hope https://x.com/stackblitz/status/1851809702793089371?s=46

**Comment by codewithtyler** [comment]
> Ok, it works for the projects created with bolt.new. It doesn't work for projects created outside of bolt.new. You cannot open some old project in bolt. You can only open in stackblitz.

I actually just created a bug for that. I actually came across the tweet today and decided to try it out. Was excited when I thought it was going to work only get the error. If it's not ready for use just yet I'm sure they'll let me know but at least I have a formal bug report.

**Comment by Phil100189** [comment]
I found a workaround. It's not perfect, but it works:
1. From bolt.new download the zip with latest progress. 
2. Upload directly ON github to repo
3. follow divpartys approach

> Just thought I would share that I was able to continue editing a **private** repo in Bolt by going to the repo ON GitHub then you will see this link in the top right which will allow you to edit it again on Bolt, hope this helps someone until they integrate it more seamlessly into their interface. ![image](https://private-user-images.githubusercontent.com/1335820/379644238-b95af18c-6df2-4ee0-a768-04122b88128e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1ODA5NTUsIm5iZiI6MTczMTU4MDY1NSwicGF0aCI6Ii8xMzM1ODIwLzM3OTY0NDIzOC1iOTVhZjE4Yy02ZGYyLTRlZTAtYTc2OC0wNDEyMmI4ODEyOGUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTAzNzM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzVmMDMzOTZhOGY0YTVmZDk5MTAzYjM5MTA1ODQwZjQ5OTE3MjE4Yzc2YjVmNDY3ZDM3MzQ5MmMxMWY1OWM5YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.tVEmcsEbmF6TXEPqJS4DWs5wabr50UItVOlh1GhXNWE)

**Comment by Labhk** [comment]
Hey, I have build this web tool for just this, previously using script on local now its a website much easy,
https://boltsync.mystify.tech/
https://www.youtube.com/watch?v=IneFM6ViV8s

**Comment by Tzalk1985** [comment]
> Hey, I have build this web tool for just this, previously using script on local now its a website much easy, https://boltsync.mystify.tech/ https://www.youtube.com/watch?v=IneFM6ViV8s

It's works nice.

**Comment by Greyvend** [comment]
We still need a way to sync private GitHub repo changes back to the bolt. That is so much wanted.

**Comment by codewithtyler** [comment]
> We still need a way to sync private GitHub repo changes back to the bolt. That is so much wanted.

The team is accepting feedback regarding GitHub and tracking that feature over on #2367

**Comment by leolux** [comment]
The current state seems to be that changes from bolt.new to stackblitz are synced but the changes made to bolt.new or stackblitz are not commited to a linked github repo after the repo has been created once from stackblitz.

**Comment by hfichter** [comment]
@codewithtyler  does that mean 2 way sync to and from github? 
- code with Bolt => commit to github
- code on my PC => push commit to github => Bolt syncs the changes
Thanks

**Comment by codewithtyler** [comment]
I'm not privy to their internal meetings but my understanding is yes they intend to add 2 way Git sync. I'm hoping it's a standard git sync and not just a GitHub sync because not everyone uses GitHub.

**Comment by Elya222** [comment]
any updates?

**Comment by codewithtyler** [comment]
They've had a couple meetings on Git (that I'm aware of) but that's all I've been told. I'll report back when I have updates.

**Comment by adambossy** [comment]
You'd think this feature wouldn't take so long if they're using the power of AI! ;)

**Comment by Shariffintech** [comment]
eagerly waiting for this feature, I have a private repo as well.

**Comment by Shariffintech** [comment]
> Hey, I have build this web tool for just this, previously using script on local now its a website much easy, https://boltsync.mystify.tech/ https://www.youtube.com/watch?v=IneFM6ViV8s

really cool, I only wish it worked with private repos :/

**Comment by KyleMoore1** [comment]
+1 this is the biggest problem with Bolt currently. A project in production needs proper version control and CI/CD

**Comment by peppix2010** [comment]
Yes, we need to be able to use version control to go forward and back with changes and the manual solution I am using now takes time and is pone to errors

**Comment by izmaelmag** [comment]
Paying 50 bucks for critical bugs not being fixed for 4 moths...
Bolt have completely ruined the project code and gives no revert options. I've uploaded the old version and can't use it again...

**Comment by peppix2010** [comment]
> Paying 50 bucks for critical bugs not being fixed for 4 months... Bolt has completely ruined the project code and gives no revert options. I've uploaded the old version and can't use it again...

It isn't very pleasant and should be solved soon, I hope.
Anyway, if you go to the project setting (bottom left) in the popup the last item is 'backups', you can restore a working version of the project from there.

Hope it helps

**Comment by TheMime3** [comment]
So is there any update on trying to import the github into bolt.new?

**Comment by codewithtyler** [comment]
@TheMime3 @kerothebosa I know there have been at least 3 meetings on adding a git integration with Bolt but that's all I know. I don't know how far along the team is on it but I know it is being worked on.

**Comment by codewithtyler** [comment]
As of today Office Hours they said we should hopefully get it in March which makes sense with it the if the month only a couple days away

**Comment by carlospliego** [comment]
Please solve this problem. Otherwise, it can't be used in our SDLC.

**Comment by peppix2010** [comment]
> As of today Office Hours they said we should hopefully get it in March which makes sense with it the if the month only a couple days away
March, of which year? :)

**Comment by careysenior** [comment]
Bump.  This issue still here - and it's March.  Create project in Bolt, export to Stackblitz, upload new repository. Make changes to Bolt code, export to Stack aaaanndd no link to Github.

**Comment by cryptekbits** [comment]
Bump. I cannot find a way to bring the project back into bolt from stackblitz. This is a major feature gap !

**Comment by nick-the-dev** [comment]
Same issue here!
Please do some update begging you

**Comment by adambossy** [comment]
Lovable supports Github sync!

**Comment by hbmartin** [comment]
> Lovable supports Github sync!

Not really, it can’t import projects not initially created in Lovable

**Comment by coopbri** [comment]
> > Lovable supports Github sync!
> 
> Not really, it can’t import projects not initially created in Lovable

Yeah, Cursor is the best I've found so far for this workflow. Early days. Excited for Bolt to implement similar!

**Comment by aaronksaunders** [comment]
I am almost at my breaking point and ready to export the whole project and load it up into cursor. I have a directory full of exported zip file as my source repo 😬

**Comment by coopbri** [comment]
> I am almost at my breaking point and ready to export the whole project and load it up into cursor. I have a directory full of exported zip file as my source repo 😬

Man, this is where I'm at right now! If you discover a better workflow in meantime let me know, will do the same.

**Comment by adambossy** [comment]
> Not really, it can’t import projects not initially created in Lovable

True. But it's also trivially easy to create an empty project in Lovable, check out your repo, and `cp -R` all the contents from your Bolt project. Then run `npm i`, `git push` and voila! You have your Bolt project running in Lovable. 

It's a one-time cost that will hopefully get fixed soon, but once you're in the Lovable ecosystem the sync works in real-time and it's fantastic.

Another perk is that Lovable also seems to write a commit history and Bolt doesn't, which is a little disappointing. I would have liked to go through old diffs and fix things Bolt did wrong.

**Comment by EddyDavies** [comment]
A great work around @adambossy and syncing existing projects is next up on their round map. The Lovable to cursor and back flow is a real gamechanger, Bolt needs to get that shipped soon. The only thing drawing me towards Bolt is their support for Expo; I can't do mobile apps on Loveable.

**Comment by tobidelly** [comment]
This is exactly what I'm experiencing at the moment.

**Comment by hbmartin** [comment]
Thanks @adambossy I tried that but now Lovable says "Read Only. If you want to update the code, set up sync with Github." 🤔 I can see my updated files in the code editor but not the preview.

**Comment by peteromano** [comment]
Fwiw, we just discovered the same issue in V0 which is wild:

1) Small business owner creates his home page in V0 and is more happy with result than using WordPress themes. 

2) I walk him through promoting/upselling to a pro vercel team with the assumption he can still use V0, while I can do more complex code tasks via traditional dev. 

3) We found that V0 will lost sync when transferring from vercel hobby to team project (V0 seems to be only for hobby projects) 

This is such a flaw, in that they're upselling, but then lose the main feature that drove the small business owner to the product in the first place. 

Finding this thread makes me think there's a general  industry issue right now trying to sync dev changes back to the AI. If vercel money doesn't support this flow either, it feels more like current Tech barrier than by design. 

I feel like in 6 months this will be figured out though.. 🤷🤔

Will check out loveable in the meantime though.

**Comment by Leohanhart** [comment]
I want to send updates from stackblits to the github repository endlessley

**Comment by epodak** [comment]
> 顺便说一句，我们刚在 V0 里发现了同样的问题，简直太离谱了：
> 
> 1. 小生意老板用 V0 做了自己的主页，感觉比用 WordPress 主题还满意，超开心！
> 2. 我跟他讲了怎么推广/推销到 Vercel 的 Pro 团队版，假设他还能继续用 V0，而我可以通过传统开发搞定更复杂的代码任务（不过从免费版跳到 40 刀/月只有 2 个座位也挺坑的，但这是另一个问题啦..）
> 3. 我们发现把 V0 从 vercel hobby 转到团队项目时居然掉线了，挺惊讶的（看来 V0 好像只适合 hobby 项目用）
> 
> 这真是个大问题，他们老是推销升级版，结果却把吸引小企业主的核心功能给搞没了。
> 
> 看到这个帖子，我觉得现在整个行业好像都面临一个问题，就是很难把开发改动同步回 AI。如果 vercel 的资金也不支持这种流程，感觉更像是当前技术瓶颈，而不是故意设计的。
> 
> 我觉得再过6个月这事儿就能搞定了吧.. 🤷🤔
> 
> 不过我先去看看《loveable》吧。

===
Fortunately, I have memberships for v0, lovable, and Replit.
